### DISCUSSION TITLE
Sessions optional persistence

### DISCUSSION THREAD

**Discussion by Somebi** [bluenviron/mediamtx#686] [General]
https://github.com/aler9/rtsp-simple-server/blob/e4ba689bfd24d496a6caa808313bb1e7ea0f3351/apidocs/openapi.yaml#L466

Cool that you made this.
Would be nice to make this persistent via config.
This way we could force all sessions to be cleared, even if something happened and some sessions were not closed properly.

**Comment by Somebi** [comment]
Providing some key/value interface to work with sessions is another idea, i.e. Redis adapter.

**Comment by Somebi** [comment]
Current implementation is returning active list of source sessions or not?

**Comment by aler9** [comment]
Hello, the problem of sessions is that clients don't reuse them once the connection with the server is broken. Therefore it's impossible to make a session persist between server restarts.

BTW, the server automatically closes not-properly-closed sessions after a while.

**Reply by Bec-k** [reply]
> Hello, the problem of sessions is that clients don't reuse them once the connection with the server is broken. Therefore it's impossible to make a session persist between server restarts.
> 
> BTW, the server automatically closes not-properly-closed sessions after a while.

You mean any rtsp server?

We are experiencing problem with connectivity to rtsp server, not yours, but inside ip camera.
It seems that there is a limit of sessions and it can't be changed.
When it reaches its limit, it just rejecting any connection.
This `after a while` is taking too long and we are forced to reboot ip camera.
That's why i was looking for a solution to properly handle sessions with source rtsp and properly TEARDOWN all sessions with source rtsp.

**Reply by Bec-k** [reply]
If we could track all source rtsp session ids, we could properly TEARDOWN them, to be sure that not a single session was not closed properly.

**Reply by aler9** [reply]
the next server version will TEARDOWN sessions at shutdown, if you want to try it immediately you can build from source by following instructions in the README.

[a comment by an automated account is not shown]
